src/Core/Commit.js

```diff
@@ -0,0 +1,331 @@
+ 			}
+
+ 			if (element.flag > ElementNode)
+ 				return
+ 	}
+
+ 	commitChildren(element, element.children, host, MountAppend, mode)
+ 	commitProps(element, element.props, PropsAppend)	
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ */
+function commitUnmount (element, parent, signature) {
+	if (element.flag === ElementComponent)
+		return componentUnmount(element, element.children, parent, signature)
+
+	commitRemove(element, parent)
+	commitDetach(element, signature)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ * @param {number} signature
+ */
+function commitReplace (element, snapshot, signature) {
+	if (signature > MountInsert && commitUnmount(element, element.parent, MountReplace))
+		return void element.state.then(function () {
+			commitReplace(element, snapshot, MountInsert)
+		})
+
+	commitMount(snapshot, elementNext(element, MountAppend), element.parent, element.host, MountInsert, ModePush)
+
+	for (var key in snapshot)
+		switch (key) {
+			case 'DOM':
+				merge(element[key], snapshot[key])
+			case 'next':
+			case 'prev':
+				break
+			default:
+				element[key] = snapshot[key]	
+		}
+}
+
+/**
+ * @param {Element} element
+ * @param {number} signature
+ */
+function commitDetach (element, signature) {
+	if (element.flag !== ElementText) {
+		var index = 0
+		var children = element.children
+		var length = children.length
+		var next = children.next
+
+		while (index++ < length)
+			switch (next.flag) {
+				case ElementComponent:
+					if (next.owner[LifecycleWillUnmount])
+						lifecycleMount(next, LifecycleWillUnmount)
+				default:
+					commitDetach(next, MountAppend)
+					next = next.next
+			}
+	}
+
+	if (element.ref)
+		commitRef(element, element.ref, RefRemove)
+
+	if (signature < MountReplace) {
+		element.context = null
+		element.state = null
+		element.event = null
+		element.DOM = null
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {(function|string)?} callback
+ * @param {number} signature
+ * @param {*} key
+ */
+function commitRef (element, callback, signature, key) {
+	switch (typeof callback) {
+		case 'string':
+			return commitRef(element, componentRef, signature, callback)			
+		case 'function':
+			switch (signature) {
+				case RefRemove:
+					return lifecycleCallback(element.host, callback, element.ref = null, key, element)
+				case RefAssign:
+					element.ref = callback
+					return lifecycleCallback(element.host, callback, element.instance || DOMTarget(element), key, element)
+				case RefReplace:
+					commitRef(element, callback, RefRemove, key)
+					commitRef(element, callback, RefAssign, key)
+			}
+			break
+		default:
+			commitRef(element, element.ref || noop, RefRemove, key)
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {string} type
+ * @param {(function|EventListener)} callback
+ */
+function commitEvent (element, type, callback) {
+	if (!element.event)
```

Comment:
Only the first event listener is ever set.

e.g.
```js
h('input', {oninput: {}, onmouseover: {}})
getEventListerns() // {input}
```

Reply:
Patched in [71def28
](https://github.com/thysultan/dio.js/pull/28/commits/71def28a6c752bfc9165c4148efb61fd1d3feb05)